oio/account/rebuilder.py

```diff
@@ -36,61 +40,78 @@ def __init__(self, conf, accounts=None, **kwargs):
 
     @staticmethod
     def string_from_item(item):
-        namespace, account = item
-        return '%s|%s' % (namespace, account)
+        namespace, account, container = item
+        if container is None:
+            return '%s|%s' % (namespace, account)
+        return '%s|%s|%s' % (namespace, account, container)
 
     def _fetch_items_from_accounts(self):
         for obj in self.accounts:
             namespace = obj['namespace']
             account = obj['account']
-            yield namespace, account
+            item = namespace, account, None
+            yield item
 
-    def _fetch_from_all_accounts(self):
+    def _fetch_items_from_all_accounts(self):
         accounts = self.account_client.account_list()
         for account in accounts:
-            yield self.namespace, account
+            item = self.namespace, account, None
+            yield item
 
     def _fetch_items(self):
         if self.accounts:
-            return self._fetch_items_from_accounts()
-
-        return self._fetch_from_all_accounts()
+            items = self._fetch_items_from_accounts()
+        else:
+            items = self._fetch_items_from_all_accounts()
+        for item in items:
+            self._accounts_to_refresh.append(item)
+            yield item
+
+        while True:
+            if not self._accounts_to_refresh:
+                break
+            item = self._accounts_refreshed.get()
+            self._accounts_to_refresh.remove(item)
```

Comment:
Since there is no guarantee on the order items will be treated, `_accounts_to_refresh` should accept fast random access, and be, for example, a set instead of a list.